include/envoy/router/router.h

```diff
@@ -164,6 +164,18 @@ class RetryPolicy {
   static const uint32_t RETRY_ON_RETRIABLE_HEADERS       = 0x1000;
   // clang-format on
 
+  /**
+   * Limitations placed on concurrent retries as a percentage of the number of active requests.
+   */
+  struct RetryBudget {
+    // The percentage of active requests that are allowed to be retries.
+    double budget_percent;
+
+    // The minimum number of active requests before enforcing the retry budget.
+    uint32_t min_concurrency;
+  };
+  using RetryBudget = struct RetryBudget;
```

Comment:
this seems unnecessary? C++ lets you refer to structs by just the name, struct qualifier is not required

Reply:
Nice. I can't believe I'm just now finding this out!